lib/smack/SmackRep.cpp

```diff
@@ -76,8 +137,20 @@ string SmackRep::type(const llvm::Type* t) {
     return BOOL_TYPE;
   else if (isFloat(t))
     return FLOAT_TYPE;
-  else
+  else if (isInt(t)) {
+    if (BIT_VECTOR) {
+      stringstream s;
+      s << "i" << getIntSize(t);
+      return s.str();
+    } else
+      return getPtrType();
+  } else if (isPointer(t))
     return getPtrType();
+  else if (t->isAggregateType())
+    return getPtrType(); 
```

Comment:
Good point! I didn't feel good when coding this function and now it is finally clean.